src/conversion/model-to-view-converters.js

```diff
@@ -311,19 +305,49 @@ export function move() {
  * @returns {Function} Remove event converter.
  */
 export function remove() {
-	return ( evt, data, conversionApi ) => {
-		const walker = new ModelTreeWalker( { boundaries: data.range, shallow: true } );
+	return ( evt, data, consumable, conversionApi ) => {
+		if ( consumable.consume( data.item, 'remove' ) ) {
+			const sourceModelRange = ModelRange.createFromPositionAndShift( data.sourcePosition, data.item.offsetSize );
+			const sourceViewRange = conversionApi.mapper.toViewRange( sourceModelRange );
 
-		let length = 0;
+			viewWriter.remove( sourceViewRange );
 
-		for ( let value of walker ) {
-			length += value.length;
+			conversionApi.mapper.unbindModelElement( data.item );
 		}
+	};
+}
 
-		const sourceModelRange = ModelRange.createFromPositionAndShift( data.sourcePosition, length );
-		const sourceViewRange = conversionApi.mapper.toViewRange( sourceModelRange );
-
-		viewWriter.remove( sourceViewRange );
+/**
+ * Function factory, creates default model-to-view converter for elements which name has changed.
+ *
+ *		modelDispatcher.on( 'rename', rename() );
+ *
+ * This converter enables default behavior of rename conversion implemented in {@link engine.conversion.ModelConversionDispatcher}.
+ * It exposes fake model element `fakeElement` and provides proper model-view bindings between original and fake model elements
+ * and their view counterparts.
+ *
+ * @external engine.conversion.modelToView
+ * @function engine.conversion.modelToView.rename
+ * @returns {Function}
+ */
+export function rename() {
+	return ( evt, data, consumable, conversionApi ) => {
+		if ( consumable.test( data.element, 'rename' ) ) {
+			// Create fake element in model, needed for conversion purposes.
```

Comment:
Reminder: in F2F talk we agreed that code from `ModelConversionDispatcher#convertRename` will be put here making the whole event more clear. The drawback is that we will need to expose `ModelConversionDispatcher` instance to the `conversionApi` or as another parameter as we will need `convertInsert`, `convertMove` and `convertRemove` methods.